rfcs/0000-kp-dashboard.md

```diff
@@ -0,0 +1,53 @@
+
+**Concrete Use Cases**
+
+1. A kpack operator upgrades his java `buildpack` to patch a CVE, which updates his `clusterstore`, which updates his `builder`. And now he would like to see that a new `build` has been triggered for _every image_ built by this java-buildpack.
+
+1. A customer engineer is brought onto a customer call. She is informed that rebase builds are not taking for a newly updated `clusterstack`. She would like to know if this behavior is happening across the board or only among select `images`. She also wants to verify that the clusterstack is configured with an active builder. She will not be familiar with any naming conventions that the customer's engineers are using.
+
+## Outcome
+
+1. The outcome is the kpack operator rarely feels the need to rely on companion CLIs (`pack`, `kubectl`) to manage their kpack workloads.
+
+1. An eventual outcome is that the `kp` tool becomes fully sufficient for debugging kpack.
+
+## Actions to take
+
+All implementation code would likely reside in the [kpack-cli](https://github.com/vmware-tanzu/kpack-cli) repository. The step to take would be to implement a `kp dashboard` command that launches a full-screen variant of the terminal interaction shown in the mockup above.
+
+## Complexity
+
+Medium.
+
+Introducing the dashboard itself into the CLI isn't more than medium complexity. Additional complexity lies in how much functionality that the dashboard will be capable of. Though this complexity can be mitigated by moving iteratively, and by playing new functionality by ear.
+
+## Prior Art
+
+**Projects**
+
+Highly interative terminal UIs have been done by projects in our domain before, notably: [dive](https://github.com/wagoodman/dive) and [k9s](https://github.com/derailed/k9s). Code for the mockup above can be found in this [./main.go](https://gist.github.com/aemengo/f8522be566ea43e44d6dae1eff6d0111).
+
+**Issues**
+
+The [problem](#problem) stated above has been echoed in previous Github Issues. Like [vmware-tanzu/kpack-cli#83](https://github.com/vmware-tanzu/kpack-cli/issues/83#issue-710376508).
+
+## Alternatives
+
+* Continue to bolster existing `kp <object> status` commands to show more information conducive to managing or debugging kpack workloads.
+* Leave this functionality out of `kp`; addressing the [problem](#problem) statement in other tooling or products.
+* Leave this functionality out of `kp`; expecting users to defer to other CLIs when needed.
+
+## Risks
+
+1. A risk lies in introducing disharmony into the user experience of the kpack-cli. An end user might wonder: "Are there are multiple ways to do the same thing? What is the best way to do X?".
```

Comment:
I think users are always looking for a suggested way to solve a problem or complete a task as a lot of the kpack resources involve new concepts (stacks, buildpacks, stores, builders, etc...).  To me it makes sense to provide some help to our users even if they could make do traversing the different resources individually.